scripts/spokepool.ts

```diff
@@ -0,0 +1,236 @@
+  ]);
+
+  // deploymentBlock = deployments[chainId.toString()].SpokePool.blockNumber;
+  // const adminAlias = ethers.utils.getAddress(ethers.BigNumber.from(admin).add(zkUtils.L1_TO_L2_ALIAS_OFFSET).toHexString());
+
+  if (chainId !== Number(spokePoolChainId)) {
+    throw new Error(`Chain ${chainId} SpokePool mismatch: ${spokePoolChainId} != ${chainId} (${spokePool.address})`);
+  }
+
+  const adminAlias = "...tbd";
+  const currentTime = Number(_currentTime);
+  const currentTimeStr = new Date(Number(currentTime) * 1000).toUTCString();
+
+  console.log(
+    `Dumping chain ${chainId} SpokePool config:\n` +
+      `\t${"HubPool chain ID".padEnd(padding)}: ${hubChainId}\n` +
+      `\t${"HubPool address".padEnd(padding)}: ${hubPoolAddress}\n` +
+      `\t${"Cross-domain admin".padEnd(padding)}: ${admin}\n` +
+      `\t${"Cross-domain alias".padEnd(padding)}: ${adminAlias}\n` +
+      `\t${"WETH".padEnd(padding)}: ${wethAddress}\n` +
+      `\t${"Current time".padEnd(padding)}: ${currentTimeStr} (${currentTime})\n`
+  );
+
+  return true;
+}
+
+function usage(badInput?: string): boolean {
+  let usageStr = badInput ? `\nUnrecognized input: "${badInput}".\n\n` : "";
+  const walletOpts = "mnemonic|privateKey";
+  const depositArgs = "--from <originChainId> --to <destinationChainId>" + " --token <tokenSymbol> --amount <amount>";
+  const dumpConfigArgs = "--chainId";
+  const fillArgs = "--from <originChainId> --hash <depositHash>";
+
+  const pad = "deposit".length;
+  usageStr += `
+    Usage:
+    \tyarn ts-node ./scripts/spokepool --wallet <${walletOpts}> ${"deposit".padEnd(pad)} ${depositArgs}
+    \tyarn ts-node ./scripts/spokepool --wallet <${walletOpts}> ${"dump".padEnd(pad)} ${dumpConfigArgs}
+    \tyarn ts-node ./scripts/spokepool --wallet <${walletOpts}> ${"fill".padEnd(pad)} ${fillArgs}
+  `.slice(1); // Skip leading newline
+  console.log(usageStr);
+
+  // eslint-disable-next-line no-process-exit
+  process.exit(badInput === undefined ? 0 : 9);
+  // not reached
+}
+
+async function run(argv: string[]): Promise<boolean> {
+  const configOpts = ["chainId"];
+  const depositOpts = ["from", "to", "token", "amount"];
+  const opts = {
+    string: ["wallet", ...configOpts, ...depositOpts],
+    default: {
+      wallet: "mnemonic",
+    },
+    unknown: usage,
+  };
+  const args = minimist(argv.slice(1), opts);
+
+  config();
+
+  let signer: Wallet;
+  switch (args.wallet) {
+    case "mnemonic":
+      signer = Wallet.fromMnemonic(process.env.MNEMONIC);
+      break;
+    case "privateKey":
+      signer = new Wallet(process.env.PRIVATE_KEY);
+      break;
+    default:
+      usage(args.wallet); // no return
+  }
+  ["MNEMONIC", "PRIVATE_KEY"].forEach((envVar) => (process.env[envVar] = ""));
+
+  switch (argv[0]) {
+    case "deposit":
+      return await deposit(args, signer);
+    case "dump":
+      return await dumpConfig(args, signer);
+    case "fill":
+      // @todo Not supported yet...
+      usage(); // no return
```

Comment:
We should add a break here for consistency otherwise it will call usage twice
```suggestion
      // @todo Not supported yet...
      usage(); // no return
      break;
```

Reply:
It won't call `usage()` because `usage()` never returns ;)

Reply:
Ended up having to do this anyway because the linter doesn't recognise that usage() never returns.